src/Riok.Mapperly/Helpers/EnumNamingStrategyHelper.cs

```diff
@@ -0,0 +1,59 @@
+using System.Globalization;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Riok.Mapperly.Abstractions;
+
+namespace Riok.Mapperly.Helpers;
+
+public static class EnumNamingStrategyHelper
+{
+    public static Dictionary<IFieldSymbol, string> BuildCustomNameStrategyMappings(
+        this ITypeSymbol enumSymbol,
+        EnumNamingStrategy namingStrategy
+    )
+    {
+        var customNameMappings = new Dictionary<IFieldSymbol, string>(SymbolEqualityComparer.Default);
+
+        var values = enumSymbol.GetMembers().OfType<IFieldSymbol>();
+        foreach (var value in values)
+        {
+            var valueString = ConvertEnumValueNameToString(value.Name, namingStrategy);
+            customNameMappings.Add(value, valueString);
+        }
+
+        return customNameMappings;
+    }
+
+    private static string ConvertEnumValueNameToString(string enumValueName, EnumNamingStrategy namingStrategy)
+    {
+        return namingStrategy switch
+        {
+            EnumNamingStrategy.PascalCase => enumValueName,
+            EnumNamingStrategy.SnakeCase => JoinWordsFrom("_", enumValueName),
+            EnumNamingStrategy.KebabCase => JoinWordsFrom("-", enumValueName),
+            _ => throw new ArgumentOutOfRangeException($"{nameof(namingStrategy)} has an unknown value {namingStrategy}"),
```

Comment:
I think it would make sense to extract these methods `ToPascalCase`, `ToSnakeCase`, `ToKebabCase` into a `NamingUtil` and add some `[Theory]` unit tests for it.

Reply:
Yes, I have moved the methods to MemberNamingUtil and added tests for them.